### DISCUSSION TITLE
Deployment of Datapipeline flink job is failing

### DISCUSSION THREAD

**Discussion by sandeep-r1** [Sunbird-Ed/Community#500] [Bugs] [answered]
Hi Team,
We are facing issue while deploying of Data pipeline flink job. Please find the error below

```
failed: [localhost] (item=ingest-router) => {
    "ansible_loop_var": "item",
    "changed": true,
    "cmd": "helm upgrade --install ingest-router /var/lib/jenkins/workspace/Deploy/alex/dev/DataPipeline/FlinkPipelineJobs/kubernetes/helm_charts/datapipeline_jobs --set job_classname=org.sunbird.dp.ingestrouter.task.IngestRouterStreamTask --set taskmanager.replicas=1 --set taskmanager.cpu_requests=0.3 -n flink-alexdev",
    "delta": "0:00:00.533824",
    "end": "2023-05-22 12:21:21.870256",
    "item": "ingest-router",
    "rc": 1,
    "start": "2023-05-22 12:21:21.336432"
}

STDOUT:

Release "ingest-router" does not exist. Installing it now.


STDERR:

Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "HorizontalPodAutoscaler" in version "autoscaling/v2beta1"


MSG:

non-zero return code
```

**Comment by santhosh-tg** [comment]
hi @sandeep-r1  Is this a new installation or a upgrade ?which release are you using ? Please add more details

**Comment by ankitach** [comment]
Here are a few steps you can try to troubleshoot and resolve the issue:

Verify the Kubernetes version: Ensure that the Kubernetes cluster you are deploying to supports the "autoscaling/v2beta1" version and that the HorizontalPodAutoscaler feature is enabled. You can check the Kubernetes documentation or consult your cluster administrator for more information.

Check the Helm chart: Review the Helm chart for the release and ensure that it is compatible with the Kubernetes version and API resources available in your cluster. Check if the chart specifies any specific versions or requirements for HorizontalPodAutoscaler.

Update Helm and Tiller: Make sure you have the latest version of Helm and Tiller (the server-side component of Helm) installed. You can update Helm using the following command:

helm repo update

Check the release status: Verify if the release "ingest-router" already exists in your cluster. You can list all the releases using the following 
helm list

If the release already exists, you can try deleting it before running the upgrade command again:
helm delete ingest-router

Note: Be cautious when deleting releases, as it may result in data loss or service disruption.

Debug Helm installation: You can enable the --debug flag in the helm upgrade command to get more detailed information about the installation process and any potential errors. This can help you identify the specific cause of the issue.

**Reply by santhosh-tg** [reply]
@sandeep-r1  Is this still an issue ? if so, please provide further information as we requested.

**Reply by srikanth-6520** [reply, marked as answer]
@santhosh-tg issue resloved , thank you.
